unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarHUDController.cs

```diff
@@ -88,6 +100,18 @@ public void Initialize(IMouseCatcher mouseCatcher, IChatController chatControlle
             chatController.OnAddMessage += OnAddMessage;
         }
 
+        if (this.sceneController != null && this.worldState != null)
+        {
+            this.sceneController.OnNewPortableExperienceSceneAdded += SceneController_OnNewPortableExperienceSceneAdded;
+            this.sceneController.OnNewPortableExperienceSceneRemoved += SceneController_OnNewPortableExperienceSceneRemoved;
+
+            List<GlobalScene> activePortableExperiences = worldState.GetActivePortableExperienceScenes();
+            foreach (GlobalScene portableExperienceToAdd in activePortableExperiences)
+            {
+                SceneController_OnNewPortableExperienceSceneAdded(portableExperienceToAdd);
+            }
```

Comment:
weren't we avoiding the use of `foreach`?

Reply:
I think the performance problem with `foreach` loops were something from the past and it was already fixed, but I'm not sure. I have made the change just in case 👍 

Maybe @pravusjif investigated it some time ago and can explain us about that?

Reply:
I personally love using `foreach` but Brian has his concerns about using it :P